.ci/no_old_refs.sh

```diff
@@ -20,7 +20,7 @@ if [ -n "$PR_HEAD_REPO_NAME" ]; then
 fi
 
 # collect issues where full link is used
-grep -IPonr "(after|[Tt]il[l]?) $GITHUB_HOST/[\w.-]+/[\w.-]+/issues/\d{1,5}" . \
+grep -IPonr "(after|[Tt]il[l]?) (<a href=\")?$GITHUB_HOST/[\w.-]+/[\w.-]+/issues/\d{1,5}" . \
```

Comment:
if  it optional , why we need this in regexp ?
lets keep regexp simple, please restore to original version.

Reply:
No violation:
`// Until https://github.com/checkstyle/checkstyle/pull/12048`

IDEA inspections violation (for javadoc):
```
/**                                                         
 * Until https://github.com/checkstyle/checkstyle/pull/12048
 */                  
```
Need to change it to:
```
/**                                                                              
 * Until <a href="https://github.com/checkstyle/checkstyle/pull/12048">#12048</a>
 */                                                                              
```

Reply:
It is used in some cases, in some, it isn't, so we need to keep it.

Reply:
.ci/no_old_refs.sh does not care about IDEA violations, it just search all urls.

if you remove change in this file, what bad will happen ?

Reply:
It cannot find `Until <a href="https://github.com/checkstyle/checkstyle/pull/12048">#12048</a>` without the modification.

> what bad will happen ?

If we have an old ref, it won't be able to determine it.

Reply:
I see